### DISCUSSION TITLE
CodePlug resets settings

### DISCUSSION THREAD

**Discussion by OE1MWW** [egzumer/uv-k5-firmware-custom#128] [General]
Loading a CodePlug will reset, disregard the settings before, Menu 33 to 'Name' and Menu 36 to '2 min'.
Not a big deal, so did anyone find other resets after CodePlug loading ?

**Comment by egzumer** [comment]
I don't know what codeplug is and how it works. This firmware extends/modifies some features. The Menu 36 has the same number of options but each option has different value. So if you set 10s in a configuration software, you will get 2min in this firmware. Menu 33 has different number of options so the software will probably reset any additional option to the nearest supported one. The same will happen with many other options. If you only use the software to upload channel setting I would recommend CHIRP as it should write back all other options as they were if you don't change them in the GUI.

**Reply by OE1MWW** [reply]
Sorry for the term 'CodePlug'. Motorola, Hytera, AnyTone, Retevis, to name a few, have a program with which you can a.) update firmware and b.) read all settings including channel frequencies, zones and all operational settings from the device or write them back into the device. And the data for b.) is called 'CodePlug' by these companies. QuanSheng also supplies a program 'PSCPS.exe'. I assume that the 'CPS' stands for 'Code Plug Setting', as with all other manufacturers. I don't know who came up with the idea of calling that set of data 'CodePLug', but that's how they call it in their manuals.

For your ingenious firmware, I am using the 0.17 at the moment (for which we are very grateful to you) we have the program PSCPS.exe for the QuanSheng UV-K5, which can read and write all frequencies and ScanList - but resets/changes some operational settings when writing. As you say, Chirp can only read and write frequencies and scan lists, but no settings. And then there is a program 'k5prog_win.exe'. This allows to read and write calibration data, writing 'Firmware' plus 'Configuration' to be read and written back to the UV-K5. It is not entirely clear what the content of the generated .bin file for 'Configuration' is. Looks like, it reads/writes all operational settings of the menues of 0.17

**Reply by egzumer** [reply]
Chirp can read and write settings, it is just the same problem like with other softwares that is meant to be used with stock FW, some settings doesn't line up. But chirp doesn't touch settings that are read and not changed in the software. Plus chirp can be used to configure channels outside of the stock bands.
